Block/Configuration.php

```diff
@@ -440,4 +438,15 @@ protected function getLandingPageConfiguration()
     {
         return $this->isLandingPage() ? $this->getCurrentLandingPage()->getConfiguration() : json_encode([]);
     }
+
+    public function canLoadInstantSearch(): bool
+    {
+        return $this->getConfigHelper()->isInstantEnabled()
+            && $this->isProductListingPage();
+    }
+
+    protected function isProductListingPage(): bool
```

Comment:
Well in this instance only `canLoadInstantSearch` is exposed to the PHTML template but right now `isProductListingPage` is only used internally within the class. 

I prefer to start with `protected` unless it's needed elsewhere or if I think it might be a good place for a plugin. In my mind, making a method public runs the risk of needing a refactor later which can have broader impact.

Reply:
That make sense. Thanks for the clarification.